serde-redis-stream-derive/src/lib.rs

```diff
@@ -75,10 +75,51 @@ pub fn redis_stream_serialize(input: TokenStream) -> TokenStream {
         .map(|field| {
             let f_ident = field.ident.expect("struct fields must have ident");
             let f_lit = f_ident.to_string();
-            let f_type = match field.ty{
-                syn::Type::Path(not_type) => not_type.path.get_ident().map(|value| value.to_owned()).unwrap(),
+            let (f_type,is_option) = match field.ty{
+                syn::Type::Path(not_type) => {                   
+                    let segments_str = &not_type.path.segments
+                    .iter()
+                    .map(|segment| segment.ident.to_string())
+                    .collect::<Vec<_>>()
+                    .join(":");                 
+
+                    let is_option = ["Option", "std:option:Option", "core:option:Option"]
+                    .iter()
+                    .any(|s| segments_str == s); 
+
+                    let temp;
+
+                    if is_option {
+                        let option_segment = ["Option", "std:option:Option", "core:option:Option"]
```

Comment:
shouldn't we be using "std::option::Option" and so on instead (double ':') for fully qualified path names (I'm not sure too)